packages/roc-package-web-app-react/app/server/router.js

```diff
@@ -33,7 +34,10 @@ export default function routes({ createRoutes, createStore, stats, dist }) {
                 this.status = 200;
                 this.body = renderPage();
             } else {
-                const store = createStore ? createStore() : null;
+                const { store, history, url } = setupForRender(createStore, this.url);
+
+                // Give Koa middlewares a chance to interact with the reduxStore
+                // This can be used to dynamically pass some data to the client.
```

Comment:
> Firstly we need to do syncHistoryWithStore before we use the history instance in match. Maybe this can be avoided if we instead manually update the store to match the whatever state we get with this method.

Hmm. You're right. Sounds a bit hacky. Is there no way to get `redux-simple-router` to dispatch the current location to the store if `history` already has a location when syncing?

> Secondly we are currently passing in the store to the function that creates the routes. I think we want this to be able to use things like onEnter to check if the user is authenticated and if not perform a redirect.

Right, so basically we want to let the eventual `onEnter`-hooks to get a reference to `getState()` and maybe even `dispatch()`?